Source/Extensions/CAShapeLayerExtension.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  CAShapeLayerExtension.swift
+//  
+//
+//  Created by Ehab Alsharkawy on 14/11/2022.
+//
+
+import UIKit
+
+extension CAShapeLayer {
+
+    /// Add Edges and corners with style
+    /// - Parameter style: Protocol that provide which edge or corner will be added with style
+    func createCustomBorder(with style: ElementBorderStyle) {
+
+        // if path not nil then it's already added so no need to draw lines or curves again
+        guard self.path == nil else { return }
+
+        // always fill is clear color
+        fillColor = UIColor.clear.cgColor
+
+        lineWidth = style.borderWidth
+        let rect = bounds.insetBy(dx: style.cornerRadius / 2,
+                                  dy: style.cornerRadius / 2)
+        let path = UIBezierPath()
+        applyBorders(with: path, rect: rect, style: style)
+        updateCorners(with: path, rect: rect, style: style)
+        self.path = path.cgPath
+    }
+
+    /*
+     (0,0)  c-----------c  (100,0)
+            |           |
+            |           |
+            |           |
+     (0,30) c-----------c  (100,30)
+
+     origin = (0,0)
+     minx = 0
+     minY = 0
+     maxX = 100
+     maxY = 30
+     */
```

Comment:
Is there any value in having this? I think it makes sense to use when writing the code, but given the unit tests and all else, would this actually help someone working on the code in the future? imo the numbers may actually end up confusing them, so we could either use x and y instead of 100 and 30, or we may even just remove the comment altogether.

What do you think?